fdbserver/workloads/ExternalWorkload.actor.cpp

```diff
@@ -100,11 +100,149 @@ struct FDBLoggerImpl : FDBLogger {
 	}
 };
 
+namespace translator {
+template <typename T>
+struct Wrapper {
+	T inner;
+};
+
+namespace metrics {
+void reserve(capi::OpaqueMetrics* c_metrics, int n) {
+	auto metrics = (std::vector<FDBPerfMetric>*)c_metrics;
+	metrics->reserve(metrics->size() + n);
+}
+void push(capi::OpaqueMetrics* c_metrics, capi::FDBMetric metric) {
+	auto metrics = (std::vector<FDBPerfMetric>*)c_metrics;
+	auto fmt = metric.fmt ? metric.fmt : "0.3g";
+	metrics->emplace_back(std::string(metric.key), metric.val, metric.avg, std::string(fmt));
```

Comment:
```suggestion
	metrics->emplace_back(FDBPerfMetric{ .name = std::string(metric.key),
	                                     .value = metric.val,
	                                     .averaged = metric.avg,
	                                     .format_code = std::string(fmt) });
```

Without this, I am seeing: 
```.../allocator_traits.h:298:9: error: no matching function for call to 'construct_at'                                                                                                                        std::construct_at(__p, std::forward<_Args>(__args)...);                                                              ^~~~~~~~~~~~~~~~~                                                                                            /usr/local/bin/../include/c++/v1/vector:808:21: note: in instantiation of function template specialization 'std::allocator_traits<std::allocator<FDBPerfMetric>>::construct<FDBPerfMetric, std::string, double &, bool &, std::string, void, void>' requested here
    __alloc_traits::construct(this->__alloc(), std::__to_address(__tx.__pos_),
                    ^
.../include/c++/v1/vector:1597:9: note: in instantiation of function template specialization 'std::vect
or<FDBPerfMetric>::__construct_one_at_end<std::string, double &, bool &, std::string>' requested here
        __construct_one_at_end(std::forward<_Args>(__args)...);
        ^
.../fdbserver/workloads/ExternalWorkload.actor.cpp:117:11: note: in instantiation of function temp
late specialization 'std::vector<FDBPerfMetric>::emplace_back<std::string, double &, bool &, std::string>' requested
here
 metrics->emplace_back(std::string(metric.key), metric.val, metric.avg, std::string(fmt));
          ^
.../include/c++/v1/__memory/construct_at.h:33:154: note: candidate template ignored: substitution failu
re [with _Tp = FDBPerfMetric, _Args = <std::string, double &, bool &, std::string>]: no matching constructor for init
ialization of 'FDBPerfMetric'
__attribute__((__visibility__("hidden"))) __attribute__((__exclude_from_explicit_instantiation__)) __attribute__((__a
bi_tag__("v15006"))) constexpr _Tp* construct_at(_Tp* __location, _Args&&... __args) {

                                    ^
1 error generated.
```

I am on `clang++ 15.06` and am passing `-stdlib=libc++ -g -std=gnu++20`. Possible that the compiler and/or standard don't need this in future versions.